include/ocpp/common/types.hpp

```diff
@@ -335,6 +335,31 @@ CaCertificateType string_to_ca_certificate_type(const std::string& s);
 /// \returns an output stream with the CaCertificateType written to
 std::ostream& operator<<(std::ostream& os, const CaCertificateType& ca_certificate_type);
 
+enum class CertificateValidationResult {
```

Comment:
I see this done before so maybe not the right time, but I think we should prevent making a copy of an enum definition at all cost. This is going to be really error prone.

If we want it in scope I would suggest using some kind of `using` statement here.

Reply:
I guess how I did it is similar to how it is typically done in libocpp. @Pietfried any suggestions?